### DISCUSSION TITLE
Getting a long "save" mp3 delay when running within a docker container (home assistant)

### DISCUSSION THREAD

**Discussion by CarlosGS** [pndurette/gTTS#281] [Q&A]
I've documented the problem in this comment: https://github.com/home-assistant/core/issues/46631#issuecomment-780425608

Could someone with gTTS experience have a look at it? I'm a bit puzzled and don't really know how to continue debugging.

There are slight differences in the dependency versions. And there's the main difference of running gTTS from a docker container within a VirtualBox VM. Thanks for any help!

**Edit 16/6:**
The issue [went away on its own](https://github.com/home-assistant/core/issues/46631#issuecomment-862182336). It must have been fixed in some Home Assistant update, or in some library update.
Your guidance was very much appreciated, thanks! :)

**Comment by pndurette** [comment]
Hi @CarlosGS!

I've been looking at your issue over there (nice investigating) but I'm really not (yet!) quite versed in how HASS works when it comes to files it generates internally, ie. regarding this bit: 

> However, on **HassOS** that `save` comand takes the much dreaded delay. **Bingo!** (I ran it through the Terminal -community- addon, and needed to `pip install gTTS` before opening python).

and—

> I've observed that the audios in `/tts/*.mp3` take a while to be generated. In fact each file only appears _exactly when the audio finally plays_.

But I'll be following to see if there's anything I can help with.

**Reply by CarlosGS** [reply]
Thank you! Something must be up with the docker container. In that respect, HASS is a mystery to me as well. But I was curious to see if you had encountered any similar problems before.

With the [python snippet](https://github.com/home-assistant/core/issues/46631#issuecomment-780425608) I have tried to replicate the internal HASS code:
https://github.com/home-assistant/core/blob/ce056656f85e46d342223f0415c5e5ae94d3466e/homeassistant/components/google_translate/tts.py#L126-L132
And it seems to be waiting at the same part - right when fetching the mp3.

Could you suggest any tool to debug that part, or a way to enable exhaustive internal logging? Thanks again!
